File: 2.py
from PrepareData import PrepareData
inputdata = PrepareData.getDataAsInt()


data = inputdata


print sum([max(line) - min(line) for line in data])

from itertools import permutations
print sum(next(x / y for x, y in permutations(line, 2) if x % y == 0) for line in data)

File: 2.scala
import scala.io.Source

object Application {

  def readLines() = {
    Source.fromFile("2.input").getLines()
  }

  def main(args: Array[String]): Unit = {
    val data = (readLines() map { _.split('\t') map { _.toInt }}).toList

    println {
        (data map { l => l.max - l.min }).sum
    }

    val r = data map { l =>
        val pairs = for { (x, idx) <- l.zipWithIndex; (y, idy) <- l.zipWithIndex; if idx != idy } yield (x, y)
        (pairs filter { case (a, b) => a % b == 0 } map { case (a, b) => a / b } take(1)).head
    }
    println(r.sum)
  }
}
